primo/opt_model/model_with_clustering.py

```diff
@@ -0,0 +1,421 @@
+#################################################################################
+# PRIMO - The P&A Project Optimizer was produced under the Methane Emissions
+# Reduction Program (MERP) and National Energy Technology Laboratory's (NETL)
+# National Emissions Reduction Initiative (NEMRI).
+#
+# NOTICE. This Software was developed under funding from the U.S. Government
+# and the U.S. Government consequently retains certain rights. As such, the
+# U.S. Government has been granted for itself and others acting on its behalf
+# a paid-up, nonexclusive, irrevocable, worldwide license in the Software to
+# reproduce, distribute copies to the public, prepare derivative works, and
+# perform publicly and display publicly, and to permit others to do so.
+#################################################################################
+
+# Standard libs
+import logging
+
+# Installed libs
+from pyomo.core.base.block import BlockData, declare_custom_block
+from pyomo.environ import (
+    Binary,
+    ConcreteModel,
+    Constraint,
+    Expression,
+    NonNegativeReals,
+    Objective,
+    Param,
+    Set,
+    Var,
+    maximize,
+)
+
+# User-defined libs
+from primo.opt_model.model_options import OptModelInputs
```

Comment:
Line 34 in model_options.py imports the PluggingCampaignModel Class from this file and line 33 here imports from model_options creating a circular dependency. I can see three options here: 1] refactor the code so that they do not depend on eachother; 2] maybe move the import inside the function that is using this class; 3] Break the circular dependency by moving the common functionality to a third file that can be imported to model_options and model_with_clustering

Reply:
Good catch Dev :-). The only reason for the import is type annotation. Best to remove it to avoid confusion